src/ui/common/src/components/tables/StickyHeaderTable.tsx

```diff
@@ -0,0 +1,121 @@
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TablePagination from '@mui/material/TablePagination';
+import TableRow from '@mui/material/TableRow';
+import Typography from '@mui/material/Typography';
+import * as React from 'react';
+
+import { Data } from '../../utils/data';
+
+interface StickyHeaderTableProps {
+  data: Data;
+}
+
+const StickyHeaderTable: React.FC<StickyHeaderTableProps> = ({ data }) => {
```

Comment:
Maybe rename this component to PaginatedTable?

While it has a sticky header, I'd argue that the pagination is more of a defining feature of this table.